metainsuranceorg.py

```diff
@@ -212,8 +212,27 @@ def enter_bankruptcy(self, time):
         self.cash = 0                           #Cash is 0 after bankruptcy.
         self.excess_capital = 0                 #Excess of capital is 0 after bankruptcy.
         self.profits_losses = 0                 #Profits and losses are 0 after bankruptcy.
+        if self.operational:
+            self.simulation.record_bankruptcy()
         self.operational = False
-        self.simulation.record_bankruptcy()
+
+
+    def market_exit(self, time):
+        [contract.dissolve(time) for contract in self.underwritten_contracts]   # removing (dissolving) all risks immediately after market exit (may not be realistic, they might instead be bought by another company)
+        self.simulation.return_risks(self.risks_kept)
+        self.risks_kept = []
+        self.reinrisks_kept = []
+        due = [item for item in self.obligations]
+        for obligation in due:
+            self.pay(obligation)
+        self.simulation.receive(self.cash)
+        self.cash = 0                           #Cash is 0 after bankruptcy.
+        self.excess_capital = 0                 #Excess of capital is 0 after bankruptcy.
+        self.profits_losses = 0                 #Profits and losses are 0 after bankruptcy.
+        if self.operational:
+            self.simulation.record_market_exit()
```

Comment:
I think this should also be recorded for proper bankruptcies entered via the `enter_bankruptcy()` method.

Reply:
It could still be good to distinguish between bankruptcies and market exits for calibration purposes.